packages/shell-api/src/error-codes.ts

```diff
@@ -0,0 +1,572 @@
+   * Signals calling `auth` of the Database API with invalid arguments.
+   *
+   * **Solution: either provide username and password as two separate arguments or provide a single object argument with `user` and `pwd` properties.**
+   */
+  DatabaseAuthArgsInvalid = 'SHAPI-10304',
+
+  /**
+   * Signals calling `auth` of the Database API with an object argument that uses the legacy `digestPassword` property.
+   *
+   * **Solution: Remove the `digestPassword` property.**
+   */
+  DatabaseAuthDigestPasswordUnsupported = 'SHAPI-10305',
+
+  /**
+   * Signals calling `createRole` of the Database API with an object that contains the `createRole` field.
+   *
+   * **Solution: Remove the `createRole` field from the object given to `createRole`.**
+   */
+  DatabaseCreateRoleNoCreateRole = 'SHAPI-10306',
+
+  /**
+   * Signals calling `printCollectionStats` of the Database API with an invalid `scale` argument.
+   *
+   * **Solution: Ensure `scale` is a number >= 1.**
+   */
+  DatabasePrintCollectionStatsScaleInvalid = 'SHAPI-10307',
+
+  /**
+   * Signals calling `enableFreeMonitoring` of the Database API on a non-primary node.
+   *
+   * **Solution: Connect to a primary node to call `enableFreeMonitoring`.**
+   */
+  DatabaseEnableFreeMonitoringNotPrimary = 'SHAPI-10308',
+
+  /**
+   * Signals calling `setProfilingLevel` of the Database API with an invalid `level`.
+   *
+   * **Solution: `level` must be in the range [0..2] (inclusive).**
+   */
+  DatabaseSetProfilingLevelLevelInvalid = 'SHAPI-10309',
+
+  /**
+   * Signals calling `setLogLevel` of the Database API with an invalid `component`.
+   *
+   * **Solution: When `component` is provided it must be of type `string`.**
+   */
+  DatabaseSetLogLevelComponentInvalid = 'SHAPI-10310',
+
+  /**
+   * Signals calling `cloneDatabase` of the Database API which has been deprecated in MongoDB 4.0 and removed.
+   *
+   * **Solution: Do not use `cloneDatabase`.**
+   */
+  DatabaseCloneDatabaseRemoved = 'SHAPI-10311',
+
+  /**
+   * Signals calling `colcloneCollection` of the Database API which has been deprecated in MongoDB 4.0 and removed.
+   *
+   * **Solution: Do not use `cloneCollection`.**
+   */
+  DatabaseCloneCollectionRemoved = 'SHAPI-10312',
+
+  /**
+   * Signals calling `copyDatabase` of the Database API which has been deprecated in MongoDB 4.0 and removed.
+   *
+   * **Solution: Do not use `copyDatabase`.**
+   */
+  DatabaseCopyDatabaseRemoved = 'SHAPI-10313',
+
+  /**
+   * Signals calling `printSecondaryReplicationInfo` of the Database API without being connected to a replica set.
+   *
+   * **Solution: Make sure you are connected to a replica set.**
+   */
+  DatabasePrintSecondaryReplicationInfoNoReplicaSet = 'SHAPI-10314',
+
+  /**
+   * Signals calling `getReplicationInfo` of the Database API without being connected to a replica set.
+   *
+   * **Solution: Make sure you are connected to a replica set.**
+   */
+  DatabaseGetReplicationInfoNoReplicaSet = 'SHAPI-10315',
```

Comment:
Maybe we could group errors where the cause and the solution is identical, e.g. all those which require being connected to a replica set under one error code?